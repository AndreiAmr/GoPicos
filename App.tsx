import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components/native';
import {
  AppThemeContext,
  AppThemeContextProvider,
} from './src/core/theme/context/theme.context';
import { theme } from './src/core/theme/theme';
import { OnBoarding } from './src/presentation/pages/OnBoarding';

const StyledComponentsTheme = () => {
  const { colorScheme } = useContext(AppThemeContext);

  return (
    <ThemeProvider theme={theme[colorScheme]}>
      <OnBoarding />
    </ThemeProvider>
  );
};

const App = () => {
  return (
    <AppThemeContextProvider>
      <StyledComponentsTheme />
    </AppThemeContextProvider>
  );
};
export default App;
