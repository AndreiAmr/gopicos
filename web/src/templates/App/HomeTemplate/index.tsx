'use client';

import { Map } from '@/atomic/atoms/Map';
import { useSidebar } from '@/components/ui/sidebar';
import React, { useEffect, useRef } from 'react';
import { Layer, Marker, Source } from 'react-map-gl/mapbox';
import * as turf from '@turf/turf';

type HomeTemplateProps = {
  spots: any[];
};

const HomeTemplate = ({ spots }: HomeTemplateProps) => {
  console.log('ðŸš€ ~ HomeTemplate ~ spots:', spots);
  const sidebar = useSidebar();

  const mapRef = useRef<any>(null);
  const containerRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (!containerRef.current) return;

    const resizeObserver = new ResizeObserver(() => {
      mapRef.current?.resize();
    });

    resizeObserver.observe(containerRef.current);

    return () => resizeObserver.disconnect();
  }, []);

  useEffect(() => {
    const handleResize = () => {
      mapRef.current?.resize();
    };

    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, [sidebar]);

  return (
    <div
      ref={containerRef}
      className=" max-w-screen  h-[calc(100%-80px)] w-full bg-background border border-white"
    >
      <Map
        ref={mapRef}
        mapboxAccessToken="pk.eyJ1Ijoic29uZWNhLWRldiIsImEiOiJjbTlwdXBvZHcwcHp3MnZvZ2dlemw1eGc2In0.eC8keYLoT3I6bGCEZrTZtg"
        initialViewState={{
          longitude: -46.6801928,
          latitude: -23.7642516,
          zoom: 13,
        }}
        onClick={(e) => {
          // console.log(e);
        }}
        mapStyle="mapbox://styles/soneca-dev/cm9pwu19h00l701s63khw175t"
        attributionControl={true}
        customAttribution={['Andrei Amaral']}
      >
        {spots.map((spot, index, array) => {
          if (!spot.coordinates || !spot.coordinates.length) {
            console.log('cai aqui no index', index, array.length);
            return null;
          }

          const feature = {
            type: 'Feature',
            properties: {},
            geometry: {
              type: 'Polygon',
              coordinates: spot.coordinates as number[][][],
            },
          } as const;

          const center = turf.center(feature).geometry.coordinates;
          console.log('ðŸš€ ~ center:', center);

          return (
            <React.Fragment key={spot.name + Date.now()}>
              <Source
                id={'location-source' + index}
                type="geojson"
                data={{
                  type: 'FeatureCollection',
                  features: [feature],
                }}
              >
                <Layer
                  id={'polygon-layer' + index}
                  type="fill"
                  paint={{
                    'fill-color': '#092178',
                    'fill-opacity': 1,
                  }}
                  key="layer-key"
                  // beforeId="water"
                />
                <Layer
                  id={'polygon-outline' + index}
                  type="line"
                  paint={{
                    'line-color': '#0b449e',
                    'line-opacity': 1,
                    'line-width': 4,
                  }}
                  key="outline-key"
                />
              </Source>
              <Source
                id={'location-point' + index}
                type="geojson"
                data={{
                  type: 'FeatureCollection',
                  features: [
                    {
                      type: 'Feature',
                      geometry: {
                        type: 'Point',
                        coordinates: center as number[],
                      },
                      properties: {},
                    },
                  ],
                }}
              >
                <Layer
                  id={'skater-layer' + index}
                  type="symbol"
                  layout={{
                    'icon-image': 'custom-skater',
                    'icon-anchor': 'center',
                    'icon-size': [
                      'interpolate',
                      ['linear'],
                      ['zoom'],
                      10,
                      0.3,
                      18,
                      1.5,
                    ],
                  }}
                />
              </Source>
            </React.Fragment>
          );
        })}
      </Map>
    </div>
  );
};

export default HomeTemplate;
