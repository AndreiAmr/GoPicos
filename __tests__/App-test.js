import React from 'react';
import 'react-native';
import App from '../App';
import { render } from '@testing-library/react-native';

describe('App', () => {
  it('hosuld render correctly', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
