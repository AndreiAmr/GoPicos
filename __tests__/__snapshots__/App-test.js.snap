// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App hosuld render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "blue",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
